### PR TITLE
Natures

### PR DESCRIPTION
**Pull request by lxgr-linux**
Added Pokete natures, that manipulate Pokete stats on an individual basis-

### CODE REVIEW

**Comment by MaFeLP** [comment]
Could you rebase this branch or merge master into it, so the review process is easier?

**Comment by lxgr-linux** [comment]
Done.

File: pokete_classes/nature.py [hunk 1]
```diff
@@ -0,0 +1,118 @@
+"""Contains everything related to the nature of Poketes, which influences some
+of their stats"""
+
+import random
+import logging
```
Comment (MaFeLP) [inline pokete_classes/nature.py:5]: Logging is not used in production

```suggestion
```

File: pokete_classes/nature.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+import scrap_engine as se
+from pokete_general_use_fns import liner
+from .ui_elements import LabelBox
+from .color import Color
+from .loops import easy_exit_loop
+
```
Comment (MaFeLP) [inline pokete_classes/nature.py:11]:
```suggestion


```

File: pokete_classes/nature.py [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+        p_n: PoketeNature object"""
+
+    def __init__(self, p_n):
+        atc = self.get_amount(p_n.nature.atc)
+        defense = self.get_amount(p_n.nature.defense)
+        init = self.get_amount(p_n.nature.initiative)
+        text = se.Text(f"Nature: {'very ' if p_n.grade == 2 else ''}") \
+               + se.Text(p_n.nature.name, esccode=Color.thicc
+                                                  + p_n.nature.esccode) \
+               + se.Text(liner(f"\n\nThat  means it has {atc} attack, \
```
Comment (MaFeLP) [inline pokete_classes/nature.py:99]:
```suggestion
            + se.Text(p_n.nature.name, esccode=Color.thicc
                      + p_n.nature.esccode) \
            + se.Text(liner(f"\n\nThat means it has {atc} attack, \
```

File: pokete_classes/nature.py [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+{defense} defense and {init} initiative then normal Poketes \
```
Comment (MaFeLP) [inline pokete_classes/nature.py:100]: Makes the sentence clearer.

```suggestion
{defense} defense and {init} initiative points more than normal Poketes \
```

File: pokete_classes/nature.py [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+
+    def get_value(self, name):
+        """Gets one attribute value by
+        ARGS:
+            name"""
+        return getattr(self.nature, name)**self.grade
+
+    def dict(self):
+        """RETURNS:
+            A dict containing inforamtion to reconstruct the object"""
```
Comment (MaFeLP) [inline pokete_classes/nature.py:67]:
```suggestion
            A dict containing information to reconstruct the object"""
```

File: pokete_classes/nature.py [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+
+    def __init__(self, nature, grade):
+        self.nature = nature
+        self.grade = grade
+        self.info = NatureInfo(self)
+
+    def get_value(self, name):
+        """Gets one attribute value by
+        ARGS:
+            name"""
```
Comment (MaFeLP) [inline pokete_classes/nature.py:62]: Missing description for `name`

File: pokete_classes/nature.py [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+            "esc": Color.green,
+            "atc": 0.9,
+            "_def": 1.1,
+        },
+        "hasty": {
+            "esc": Color.red,
+            "_def": 0.9,
+            "init": 1.1
+        },
+        "normal": {}
```
Comment (MaFeLP) [inline pokete_classes/nature.py:51]: I would put these into `pokete_data`, as this dict does not hold logic, but data.

Putting it into `pokete_data` also means, mods can modify these values or add new natures.

File: pokete_classes/nature.py [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+        self.atc = atc
+        self.defense = _def
+        self.initiative = init
+
+
+class PokeNature:
+    """Holds specific information about a Pokete
+    ARGS:
+        nature: The Nature of the Pokete
+        grade: The Natures grade, 1 or 2"""
```
Comment (MaFeLP) [inline pokete_classes/nature.py:33]:
```suggestion
        grade: The Nature's grade, 1 or 2"""
```

**Review by MaFeLP — CHANGES_REQUESTED** [review]

File: pokete_classes/nature.py (hunk 7 above)
Comment (lxgr-linux) [reply pokete_classes/nature.py:51]: That's actually a very good point.

**Review by MaFeLP — APPROVED** [review]

**Pull request merged**